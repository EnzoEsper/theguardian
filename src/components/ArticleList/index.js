import React from 'react';

const ArticleList = ({articles}) => {

  return(
      <ul className="ui list">
        {articles.map(({ webUrl, webTitle }) => (
          <li key={webUrl}>
            <a href={webUrl} target="_blank" rel="noopener noreferrer">
              {webTitle}
            </a>
          </li>
        ))}
      </ul>
  );
};

export default ArticleList;