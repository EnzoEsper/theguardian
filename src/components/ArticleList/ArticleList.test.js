import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

import ArticleList from './';

Enzyme.configure({
  adapter: new EnzymeAdapter(),
});

// Mock de datos para pasar al componente ArticleList
const mockArticles = [
  {
    webUrl: "https://google.com",
    webTitle: "This is google"
  }
];

describe("Article List component", () => {

  // Test para comporbar si el componente es renderizado, esta vez usando snapshot
  //Cada vez que se ejecuta este test, compara el DOM actual con el existente en la carpeta
  // __snapshots__
  it("Renders without errors ", () => {
    const wrapper = shallow(<ArticleList articles={mockArticles}/>);

    expect(wrapper).toMatchSnapshot();
  });

  // Test para comprobar que retorne el arreglo vacio por defecto cuando no se pasen datos
  it("Returns the default empty array when there is no data to map trough", () => {
    const wrapper = shallow(<ArticleList articles={[]}/>);

    expect(wrapper).toMatchSnapshot();
  });

  // Test para comprobar que el componente no se rompe incluso cuando se le pasa como prop el // arreglo de articulos vacio, se hace chequeando que no hay etiquetas <li> (ya que, en  
  // caso de haber algun articulo/datos, si se procesarian y habrian etiquetas <li> ) 
  it("Doesn't break without articles", ()=> {
    const wrapper = shallow(<ArticleList articles={[]}/>);

    expect(wrapper.find('li')).toHaveLength(0);
  });

  // Comprueba que el componente presenta los resultados de bÃºsqueda cuando los articulos/el estado cambia
  it("Renders search results when the articles change", () => {
    const wrapper = mount(<ArticleList articles={[]} />);

    wrapper.setProps({
      articles: [{ webUrl: "http://google.com", webTitle: "Google Search" }]
    });

    // console.log(wrapper.debug());
    expect(wrapper.find("a").prop("href")).toEqual("http://google.com");
  });


});

