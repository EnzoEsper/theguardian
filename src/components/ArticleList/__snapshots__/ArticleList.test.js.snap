// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List component Renders without errors  1`] = `ShallowWrapper {}`;

exports[`Article List component Returns the default empty array when there is no data to map trough 1`] = `ShallowWrapper {}`;
